libyul/AsmAnalysis.cpp

```diff
@@ -515,11 +521,9 @@ void AsmAnalyzer::expectBoolExpression(Expression const& _expr)
 		m_errorReporter.typeError(
 			1733_error,
 			locationOf(_expr),
-			"Expected a value of boolean type \"" +
-			m_dialect.boolType.str() +
-			"\" but got \"" +
-			type.str() +
-			"\""
+			"Expected a value of boolean type {:q} but got {:q}",
+			m_dialect.boolType,
+			type
```

Comment:
> Not sure if adding fmtlib is worth it just for this but for me it does improve overall readability.

Actually, that was exactly the motivation for me to add that. I am fine in not doing so though, but I hoped for getting it in.

Sadfy we cannot get string interpolation in C++, generally speaking, so that solution would still be best for lowering the noise when doing formatted string composition.

> One thing I do not like is that the parameters go to typeError() instead of the format function. I think that an explicit call to fmt::format() might be better.

Well, that's easy: just call it like that (I just thought people would have liked it integrated more than explicit):

```cpp
m_errorReporter.typeError(...., fmt::format("Your message {here}.", fmt::arg("here", someHereObject));
```

Reply:
> (I just thought people would have liked it integrated more than explicit)

For me the problem is that `typeError()` has lot of other parameters already so it's not very clear that the string is special and all the ones following it are just format parameters. An explicit `fmt::format()` makes it more apparent.

Reply:
@cameel fine with me. :)